### PR TITLE
Flags for client methods, support for weak clients

### LINKED ISSUE #3: Support for weak clients
There's currently no support for weak clients in MPV#.

Docs for [`mpv_create_weak_client`](https://github.com/mpv-player/mpv/blob/release/0.35/libmpv/client.h#L574):

```
/**
 * This is the same as mpv_create_client(), but the created mpv_handle is
 * treated as a weak reference. If all mpv_handles referencing a core are
 * weak references, the core is automatically destroyed. (This still goes
 * through normal uninit of course. Effectively, if the last non-weak mpv_handle
 * is destroyed, then the weak mpv_handles receive MPV_EVENT_SHUTDOWN and are
 * asked to terminate as well.)
 *
 * Note if you want to use this like refcounting: you have to be aware that
 * mpv_terminate_destroy() _and_ mpv_destroy() for the last non-weak
 * mpv_handle will block until all weak mpv_handles are destroyed.
 */
```

### LINKED ISSUE #4: Create enum flags for Client methods
Client has methods like `Seek` that support flags. Currently flags are set as strings and should match flags from mpv manual. This can lead to unexpected crashes in case of typos or wrong flags.
Using enum flags will make these methods less error prone.

### PR DESCRIPTION
**Pull request by fsobolev**
Closes #4 
Closes #3

### CODE REVIEW

File: Nickvision.MPVSharp/Internal/MPVClient.cs [hunk 1]
```diff
@@ -115,33 +117,38 @@ public partial class MPVClient : ICloneable
 
     private const int LC_NUMERIC = 1;
 
-    protected readonly nint _handle;
-
+    /// <summary>
+    /// MPV client handle
+    /// </summary>
+    public nint Handle { get; }
```
Comment (nlogozzo) [inline Nickvision.MPVSharp/Internal/MPVClient.cs:123]: { get; init; }

**Review by nlogozzo — CHANGES_REQUESTED** [review]
Can you fix the examples too?

Edit: just saw the new commit :)

File: Nickvision.MPVSharp/Internal/MPVClient.cs (hunk 1 above)
Comment (fsobolev) [reply Nickvision.MPVSharp/Internal/MPVClient.cs:123]: isn't it redundant?

Comment (nlogozzo) [reply Nickvision.MPVSharp/Internal/MPVClient.cs:123]: No because I believe it resorts to { private set; } by default

Comment (fsobolev) [reply Nickvision.MPVSharp/Internal/MPVClient.cs:123]: ah okay

**Comment by nlogozzo** [comment]
Will merge when I get home from work and release next ver :)

**Review by nlogozzo — APPROVED** [review]

**Comment by nlogozzo** [comment]
https://www.nuget.org/packages/Nickvision.MPVSharp/2023.7.1

**Pull request merged**